students/scott_bromley/lesson09/mailroom_oo/donor_models.py

```diff
@@ -0,0 +1,264 @@
+        return self.donor_name.split()[1]
+
+    @property
+    def num_donations(self):
+        """
+        getter for number of donations
+        :return: number of donations
+        """
+        return len(self.donations)
+
+    @property
+    def total_donations(self):
+        """
+        getter for sum of donations
+        :return: donations sum for donor
+        """
+        return sum(self.donations)
+
+    @property
+    def avg_donation(self):
+        """
+        getter for average donation
+        :return: average donation of donor
+        """
+        if self.num_donations == 0:
+            raise ValueError(f"{self.donor_name} has no donations")
+        return round(mean(self.donations), 2)
+
+    def add_donation(self, donation):
+        """
+        adds donation to Donor donations
+        :param self:
+        :param donation: donation amount
+        :return: None
+        """
+        if donation <= 0:
+            raise ValueError("invalid donation amount")
+        else:
+            try:
+                self.donations.append(float(donation))
+            except TypeError as type_err:
+                raise type_err("invalid donation type")
+
+    def thank_you(self):
+        """
+        create formatted letter string for donor
+        :return: formatted letter
+        """
+        if not self._donations:
+            raise ValueError("donor has no donations")
+        else:
+            return f"\n\n{datetime.today().strftime('%Y-%d-%m')},\n\n\n" \
+                   f"Dear {self.donor_name},\n\n\n" \
+                   f"On behalf of all of us we thank you for your generous\n" \
+                   f"donation of ${self.donations[-1]:03.2f}.\n" \
+                   f"To date, you have donated: ${self.total_donations:03.2f}.\n" \
+                   "We look forward to seeing you at our annual meeting.\n\n" \
+                   "Sincerely,\n\n" \
+                   "ADL\n"
+
+    def report_row(self):
+        """
+        format report row for Donor object
+        :return: formatted report row Donor
+        """
+        if not self._donations:
+            raise ValueError("donor has no donations")
+        else:
+            return ('{:<25}{:^5}${:>14,.2f}{:^5}{:>10}{:^5}${:>14,.2f}'
+                    ).format(self.donor_name, ' ',
+                             self.total_donations, ' ',
+                             self.num_donations, ' ',
+                             self.avg_donation
+                             )
+
+    def __str__(self):
+        """
+        prints Donor object
+        :return: String representation of Donor object
+        """
+        return f"Donor: {self._donor_name}, Donations: {self._donations}"
+
+    def __repr__(self):
+        """
+        :return: string of self instantiation
+        """
+        return f"Donor({self._donor_name!r}, {self._donations!r})"
+
+    @classmethod
+    def sort_key(cls):
+        return cls.last_name, cls.first_name, cls.donations
```

Comment:
name and donations are not class attributes - sort_key should be an instance method, not classmethod.